riscv-pages/src/state.rs

```diff
@@ -111,3 +111,18 @@ impl InternalData for InternalDirty {}
 impl Cleanable for InternalDirty {
     type Cleaned = InternalClean;
 }
+
+/// A shared (and implicitly dirty) page that has been assigned as a mapped page.
+#[derive(Debug)]
+pub enum SharedDirty {}
+
+/// A mapped page that is eligible for conversion to a Shared page.
+#[derive(Debug)]
+pub enum Shareable {}
+impl State for Shareable {}
+impl Converted for Shareable {}
+impl State for SharedDirty {}
+impl Mappable<MeasureOptional> for SharedDirty {}
+impl Assignable<MeasureOptional> for Shareable {
```

Comment:
I don't think `Assignable` makes sense here; ownership isn't changing for shared pages.